apps/ledger-live-desktop/src/renderer/components/SignMessageConfirm/index.jsx

```diff
@@ -55,43 +59,68 @@ type Props = {
   signMessageRequested: TypedMessageData | MessageData,
 };
 
-const SignMessageConfirm = ({ device, account, signMessageRequested: message }: Props) => {
+const SignMessageConfirm = ({ device, account, signMessageRequested }: Props) => {
   const type = useTheme("colors.palette.type");
   const { t } = useTranslation();
 
   if (!device) return null;
+  const inferredNanoFields = (() => {
+    try {
+      if (account.currency.family === "ethereum") {
+        const parsedMessage =
+          typeof signMessageRequested.message === "string"
+            ? JSON.parse(signMessageRequested.message)
+            : signMessageRequested.message;
 
-  const fields = [];
+        return isEIP712Message(signMessageRequested.message)
+          ? getNanoDisplayedInfosFor712(parsedMessage)
+          : null;
+      }
+      throw new Error();
+    } catch (e) {
+      return null;
+    }
+  })();
 
-  if (message.hashes && message.hashes.domainHash) {
-    fields.push({
+  let fields = [];
+  if (Array.isArray(inferredNanoFields)) {
+    fields = inferredNanoFields.map(field => ({
+      ...field,
       type: "text",
-      label: t("SignMessageConfirm.domainHash"),
-      // $FlowFixMe
-      value: message.hashes.domainHash,
-    });
-  }
-  if (message.hashes && message.hashes.messageHash) {
-    fields.push({
-      type: "text",
-      label: t("SignMessageConfirm.messageHash"),
-      // $FlowFixMe
-      value: message.hashes.messageHash,
-    });
-  }
-  if (message.hashes && message.hashes.stringHash) {
+      value: Array.isArray(field.value) ? field.value.join(",\n") : field.value,
+    }));
+  } else {
+    const { hashes, message } = signMessageRequested;
+    if (hashes && hashes.domainHash) {
+      fields.push({
+        type: "text",
+        label: t("SignMessageConfirm.domainHash"),
+        // $FlowFixMe
```

Comment:
no way to convert this file to tsx ?

Reply:
I could, but it would probably imply doing a lot more changes (to other files mostly) in this PR just to make all the types correct, which would create a lot more noise in a pretty already complicated PR.